pg_search/src/bootstrap/create_bm25.rs

```diff
@@ -489,3 +492,33 @@ fn index_size(index_name: &str) -> Result<i64> {
 
     Ok(total_size as i64)
 }
+
+fn add_pg_depend_entry(index_oid: pg_sys::Oid, schema_oid: pg_sys::Oid) {
+    unsafe {
+        let index_dep = pg_sys::ObjectAddress {
+            classId: pg_sys::RelationRelationId,
+            objectId: index_oid,
+            objectSubId: 0,
+        };
+
+        let schema_dep = pg_sys::ObjectAddress {
+            classId: pg_sys::NamespaceRelationId,
+            objectId: schema_oid,
+            objectSubId: 0,
+        };
+
+        // Create dependency from index to schema
+        pg_sys::recordDependencyOn(
+            &index_dep,
+            &schema_dep,
+            pg_sys::DependencyType::DEPENDENCY_NORMAL,
+        );
+
+        // Create dependency from schema to index
+        pg_sys::recordDependencyOn(
+            &schema_dep,
+            &index_dep,
+            pg_sys::DependencyType::DEPENDENCY_NORMAL,
+        );
```

Comment:
@neilyio, try adding a `pg_sys::CommandCounterIncrement();` at the end of this function.

Reply:
I pushed up so you could take a look, but seeing the same errors.

Reply:
There's nothing wrong with this code (CommandCounterIncrement() may or may not be necessary in the end) that I can see, so I wonder if this is instead highlighting a problem somewhere else?

If you can repro this via psql, it'd be nice to get a backtrace so we can see the call stack to the error -- it's not clear what actual bit of code is causing it.

Reply:
Running in the debugger (haven't been able to get a backtrace to show up), I can confirm the `uuid` test fails at `src/bootstrap/create_bm25.rs:447` in the `Spi::run` when we `EXECUTE 'DROP INDEX IF EXISTS {}.{} CASCADE';` with:
```
ERROR:  could not open relation with OID 1247453
```

The `bm25_partial_index_alter_and_drop` test doesn't seem to throw directly in our extension code, but the client is sending `ALTER TABLE paradedb.test_partial_index DROP COLUMN category CASCADE;` on `tests/bm25_search.rs:1090` with:
```
cannot DROP TABLE \"test_partial_index\" because it is being used by active queries in this session
```

Reply:
Actually I think I see where we're getting confused here, we are indeed referencing the wrong schema.

The `pg_relation.namespace_oid()` call is reference the schema of the _table_ on which the index being created. That's not what we're trying to drop.

We're trying to drop the generated schema that contains the `search` method. I think for that reason, we do need to be looking up with this code:

```rust
  let schema_oid_query = format!(
        "SELECT oid FROM pg_namespace WHERE nspname = {}",
        spi::quote_literal(index_name)
    );
    let schema_oid = Spi::get_one::<pg_sys::Oid>(&schema_oid_query)
        .expect("error looking up schema in create_bm25")
        .expect("no oid for schema created in create_bm25")
        .as_u32();
```

Reply:
Is there a `struct PgRelation` instance somewhere in the `create_bm25()` function or the `drop_bm25()` function that hasn't been dropped?  I don't see anything obvious.  I dunno.  

Spi is really gross and can cause weird shenanigans

Reply:
I'm not sure if we're talking about the same thing here, I might need to clarify what I was trying above.

We _are not_ trying to register a dependency between the _table_'s schema and the index. We _are_ trying to register a dependency between the _newly-created_ schema and the index. That's why the OID that we're retrieving with `pg_relation.namespace_oid()` is not the OID of the correct schema.

This "newly-created schema" is the one from the `CREATE SCHEMA` call on `src/bootstrap/create_bm25.rs`. I need to get the OID of _that_ schema so I can use it in the `recordDependencyOn` call.

That's why I believe we should indeed be retrieving that OID like so:

```rust
    let schema_oid_query = format!(
        "SELECT oid FROM pg_namespace WHERE nspname = {}",
        spi::quote_literal(index_name)
    );
    let schema_oid = Spi::get_one::<pg_sys::Oid>(&schema_oid_query)
        .expect("error looking up schema in create_bm25")
        .expect("no oid for schema created in create_bm25")
        .as_u32();
```

Reply:
> We are trying to register a dependency between the newly-created schema and the index

yes. Duh. You are absolutely right. I had this wrong in my head. 

Sorry for the noise and carry on!